pkg/election/election.go

```diff
@@ -246,14 +246,15 @@ func (e *Election) campaignLoop(ctx context.Context, session *concurrency.Sessio
 		elec := concurrency.NewElection(session, e.key)
 		ctx2, cancel2 := context.WithCancel(ctx)
 
+		campaignWg.Add(1)
```

Comment:
does this change has some effects?

Reply:
Probably not. This is a problem that I found by reading code, but the probability of this problem triggering is very small.

Reply:
@lance6716  If we evict leader before line 246, can campaign be canceled successfully?

Reply:
then in line 260, the atomic variable loaded `false` and the campaign goroutine will not work